crates/common/certificate/src/lib.rs

```diff
@@ -105,23 +105,16 @@ pub struct KeyCertPair {
     certificate: Zeroizing<rcgen::Certificate>,
 }
 
-impl KeyCertPair {
-    pub fn new_selfsigned_certificate(
-        config: &NewCertificateConfig,
-        id: &str,
-        key_kind: &KeyKind,
-    ) -> Result<KeyCertPair, CertificateError> {
-        let today = OffsetDateTime::now_utc();
-        let not_before = today - Duration::days(1); // Ensure the certificate is valid today
-        KeyCertPair::new_selfsigned_certificate_at(config, id, not_before, key_kind)
-    }
+pub struct CertParameters {
+    params: CertificateParams,
+}
```

Comment:
[Minor] I would avoid to add this type.

What you need are simple functions to create and populate `rcgen::CertificateParams` for different usages.